pkg/commands/git_commands/working_tree.go

```diff
@@ -49,30 +46,37 @@ func (self *WorkingTreeCommands) StageFiles(paths []string) error {
 	quotedPaths := slices.Map(paths, func(path string) string {
 		return self.cmd.Quote(path)
 	})
-	return self.cmd.New(fmt.Sprintf("git add -- %s", strings.Join(quotedPaths, " "))).Run()
+
+	cmdStr := NewGitCmd("add").Arg("--").Arg(strings.Join(quotedPaths, " ")).ToString()
+
+	return self.cmd.New(cmdStr).Run()
 }
 
 // StageAll stages all files
 func (self *WorkingTreeCommands) StageAll() error {
-	return self.cmd.New("git add -A").Run()
+	cmdStr := NewGitCmd("add").Arg("-A").ToString()
+
+	return self.cmd.New(cmdStr).Run()
 }
 
 // UnstageAll unstages all files
 func (self *WorkingTreeCommands) UnstageAll() error {
-	return self.cmd.New("git reset").Run()
+	return self.cmd.New(NewGitCmd("reset").ToString()).Run()
 }
 
 // UnStageFile unstages a file
 // we accept an array of filenames for the cases where a file has been renamed i.e.
 // we accept the current name and the previous name
 func (self *WorkingTreeCommands) UnStageFile(fileNames []string, reset bool) error {
-	command := "git rm --cached --force -- %s"
-	if reset {
-		command = "git reset HEAD -- %s"
-	}
-
 	for _, name := range fileNames {
-		err := self.cmd.New(fmt.Sprintf(command, self.cmd.Quote(name))).Run()
+		var cmdStr string
+		if reset {
+			cmdStr = NewGitCmd("reset").Arg("HEAD").Arg("--").Arg(self.cmd.Quote(name)).ToString()
```

Comment:
if we state the command on one line it's not too hard to read, but given the realestate being taken up by the method calls, you can't fit much on one line

Reply:
Maybe we could also have `Args(flags ...string)` for cases like these?

Although, then we wouldn't really need `Arg(flag string)`, would we?

Reply:
I like that. So it would become 
```go
NewGitCmd("reset").Args("HEAD", "--", self.cmd.Quote(name)).ToString()
```
I'm fine with having both `Args` and `Arg`.